File: Examples/Ruby/Images/post_replace_image.rb
require 'aspose_pdf_cloud'

class Image

  include AsposePDFCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @pdf_api = PdfApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def post_replace_image
    file_name = "SampleImage.pdf"
    upload_file(file_name)
    image_file_name = "aspose-cloud.png"
    upload_file(image_file_name)

    page_number = 1 
    image_number = 1
    response = @pdf_api.post_replace_image(file_name, page_number, image_number, {image_file: image_file_name})
  end

end

image_obj = Image.new()
puts image_obj.post_replace_image

File: Examples/PHP/Images/PostReplaceImage.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\PDF\PdfApi;
use Aspose\PDF\AsposeApp;

class Images {

	public $pdfApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->pdfApi = new PdfApi();
	}

	public function postReplaceImage() {
		// Upload file to Aspose Cloud Storage
		$fileName = "SampleImage.pdf";
		Utils::uploadFile($fileName);

		$file = realpath(__DIR__ . '/../../..') . '/Data/aspose-cloud.png';
		$result = $this->pdfApi->PostReplaceImage($fileName, $pageNumber = 1, $imageNumber = 1, $imageFile = "", $storage = "", $folder = "", $file);
		print_r ( $result );
	}
}

$images = new Images();
$images->postReplaceImage();

?>
//ExEnd:
